File: VB/WindowsFormsApplication25/Form1.vb
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.Xpo
Imports DevExpress.XtraGrid.Views.Grid

Namespace WindowsFormsApplication25

    Public Partial Class Form1
        Inherits Form

        Private uow As UnitOfWork

        Public Sub New()
            InitializeComponent()
            uow = New UnitOfWork()
            Dim xpc As XPCollection(Of DomainObject) = New XPCollection(Of DomainObject)(uow)
            gridControl1.DataSource = xpc
            AddHandler gridView1.RowStyle, New RowStyleEventHandler(AddressOf gridView1_RowStyle)
            AddHandler gridControl1.ViewRegistered, New DevExpress.XtraGrid.ViewOperationEventHandler(AddressOf gridControl1_ViewRegistered)
        End Sub

        Private Sub gridControl1_ViewRegistered(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.ViewOperationEventArgs)
            If TypeOf e.View Is GridView Then
                AddHandler CType(e.View, GridView).RowStyle, New RowStyleEventHandler(AddressOf Form1_RowStyle)
            End If
        End Sub

        Private Sub Form1_RowStyle(ByVal sender As Object, ByVal e As RowStyleEventArgs)
            Dim gridView As GridView = CType(sender, GridView)
            If Not gridView.IsDataRow(e.RowHandle) Then Return
            Dim data As PropertyData = TryCast(gridView.GetRow(e.RowHandle), PropertyData)
            If data IsNot Nothing AndAlso Not Equals(data.NewValue, data.OldValue) Then
                e.Appearance.BackColor = Color.LightSalmon
            End If
        End Sub

        Private Sub gridView1_RowStyle(ByVal sender As Object, ByVal e As RowStyleEventArgs)
            Dim gridView As GridView = CType(sender, GridView)
            If Not gridView.IsDataRow(e.RowHandle) Then Return
            If uow.IsObjectToSave(gridView.GetRow(e.RowHandle)) Then
                e.Appearance.BackColor = Color.LightSalmon
            End If
        End Sub

        Private Sub gridControl1_EmbeddedNavigator_ButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.NavigatorButtonClickEventArgs)
            uow.CommitChanges()
        End Sub
    End Class
End Namespace

File: CS/WindowsFormsApplication25/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.Xpo;
using DevExpress.XtraGrid.Views.Grid;

namespace WindowsFormsApplication25 {
    public partial class Form1 : Form {

        private UnitOfWork uow;
        public Form1() {
            InitializeComponent();
            uow = new UnitOfWork();
            XPCollection<DomainObject> xpc = new XPCollection<DomainObject>(uow);
            gridControl1.DataSource = xpc;

            gridView1.RowStyle += new DevExpress.XtraGrid.Views.Grid.RowStyleEventHandler(gridView1_RowStyle);
            gridControl1.ViewRegistered += new DevExpress.XtraGrid.ViewOperationEventHandler(gridControl1_ViewRegistered);
        }

        void gridControl1_ViewRegistered(object sender, DevExpress.XtraGrid.ViewOperationEventArgs e) {
            if (e.View is GridView) {
                ((GridView)e.View).RowStyle += new RowStyleEventHandler(Form1_RowStyle);
            }
        }

        void Form1_RowStyle(object sender, RowStyleEventArgs e) {
            GridView gridView = (GridView)sender;
            if (!gridView.IsDataRow(e.RowHandle)) return;
            PropertyData data = gridView.GetRow(e.RowHandle) as PropertyData;
            if (data != null && !Equals(data.NewValue, data.OldValue)) {
                e.Appearance.BackColor = Color.LightSalmon;
            }
        }

        void gridView1_RowStyle(object sender, DevExpress.XtraGrid.Views.Grid.RowStyleEventArgs e) {
            GridView gridView = (GridView)sender;
            if (!gridView.IsDataRow(e.RowHandle)) return;
            if (uow.IsObjectToSave(gridView.GetRow(e.RowHandle))) {
                e.Appearance.BackColor = Color.LightSalmon;
            }
        }

        private void gridControl1_EmbeddedNavigator_ButtonClick(object sender, DevExpress.XtraEditors.NavigatorButtonClickEventArgs e) {
            uow.CommitChanges();
        }
    }
}
